### DISCUSSION TITLE
The Alt Text flag seems backward

### DISCUSSION THREAD

**Discussion by mikehamca** [mastodon/mastodon#28401] [General]
Currently, images have a flag, "ALT," when the image has alt text present. If the image lacks alt text, no flag is displayed.

This seems backward to me. The normal (almost universal) situation is that images in Mastodon _do_ have alt text because almost all Mastodon users, aware that some users need alt text, supply it. So the common situation is flagged, the exceptions are not. 

This is like wanting to identify which trees in a forest are dead, and deciding that the best way to do that is to put a sign, "ALIVE," on every living tree, and leave dead trees unmarked. 

Wouldn't it be better to have no flag on images that do have alt text (since those are the standard, usual images), and put a flag — say, "NO ALT" — on images that lack alt text?

**Comment by mjankowski** [comment]
I think its mainly there to give you a way to reveal it ... unless I don't understand what you are saying. In that case can you add an image with some indication of what you think is wrong and how it could be made more right?

> This is like wanting to identify which trees in a forest are dead, and deciding that the best way to do that is to put a sign, "ALIVE," on every living tree, and leave dead trees unmarked.

That would be a totally exhausting thing to do. Because you know what would happen -- as soon as you are done there'd be new baby trees coming up and you'd have to go tag them again.

**Comment by mikehamca** [comment]
Thanks for replying. I just posted an image twice, once without alt text and once with:

Without:
[in the email, an image of a full moon luminous behind some evening clouds; there's a label in the upper right corner: "Hide."]

With:
[in the email, the same image of the moon, but in addition to the "Hide" label, there's a label in the bottom right corner: "Alt."]

As you see, there’s no comment on the image *without* alt text — the fact that the image lacks alt text flies under the radar, as it were, leaving the image unblemished to the eyes of those who can see but worthless for those who cannot.

But the image *with* alt text has a prominent label to tell those who don’t need alt text that alt text is present, a label invisible to those who do need alt text.

My suggestion is to present images that have alt text unblemished — without a label, since those images are (I hope) the norm. But images *lacking* alt text should be labeled, perhaps with "Alt Text Missing." That label would irritate those who see it, but that’s only fair, since the omission of alt text doubtless irritates those who cannot see.

The basic idea: images with alt text should be satisfactory for all; images without alt text should be somewhat of an irritant to all.

Hope this helps. And thanks again for seeking clarification.

Michael
